### PR TITLE
Add voice broadcast recording body

### PR DESCRIPTION
**Pull request by weeman1337**
Part of https://github.com/vector-im/element-web/issues/23282

This PR contains some components to be used in voice broadcast as well as event enums and interfaces.

vb

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: res/css/voice-broadcast/molecules/_VoiceBroadcastRecordingBody.pcss [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+.mx_VoiceBroadcastRecordingBody {
+    align-items: flex-start;
+    background-color: $quinary-content;
+    border-radius: 8px;
+    display: inline-flex;
+    gap: 8px;
```
Comment (kerryarchibald) [inline res/css/voice-broadcast/molecules/_VoiceBroadcastRecordingBody.pcss:22]: Can these use spacing variables?

File: res/css/voice-broadcast/molecules/_VoiceBroadcastRecordingBody.pcss [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    padding: 12px;
+}
+
+.mx_VoiceBroadcastRecordingBody_title {
+    font-size: 12px;
```
Comment (kerryarchibald) [inline res/css/voice-broadcast/molecules/_VoiceBroadcastRecordingBody.pcss:27]: font variable

File: src/components/atoms/Icon.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        `mx_Icon_${size}`,
+        `mx_Icon_${colour}`,
+    ];
+
+    const styles: React.CSSProperties = {
+        maskImage: `url("${iconTypeMap.get(type)}")`,
+    };
+
+    return (
+        <div
```
Comment (kerryarchibald) [inline src/components/atoms/Icon.tsx:60]: Maybe this should use an `i` element? It should also probably set `role="presentation"` and `aria-hidden` with rest props support to override them.

File: src/components/atoms/Icon.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    Live,
+}
+
+const iconTypeMap = new Map([
+    [IconType.Live, liveIcon],
+]);
+
+export enum IconColour {
+    Accent = "accent",
+    White = "white",
```
Comment (kerryarchibald) [inline src/components/atoms/Icon.tsx:31]: How does this work with themes?

File: src/components/views/messages/MessageEvent.tsx [hunk 5]
```diff
@@ -74,6 +75,7 @@ const baseEvTypes = new Map<string, React.ComponentType<Partial<IBodyProps>>>([
     [M_POLL_START.altName, MPollBody],
     [M_BEACON_INFO.name, MBeaconBody],
     [M_BEACON_INFO.altName, MBeaconBody],
+    [VoiceBroadcastInfoEventType, VoiceBroadcastBody],
```
Comment (kerryarchibald) [inline src/components/views/messages/MessageEvent.tsx:78]: It seems like we mostly use upper camel case for constant naming in js-sdk, but it's quite vague in the code style doc. I'm not really sure what 'general naming' is
> 9. lowerCamelCase is used for function and variable naming.
> 10. UpperCamelCase is used for general naming.

Comment (kerryarchibald) [reply src/components/views/messages/MessageEvent.tsx:78]: Anyway, more a comment for the code style docs than this PR

Comment (weeman1337) [reply src/components/views/messages/MessageEvent.tsx:78]: I am not sure either :laughing: 

If 9. mentions functions and variables then everything else is 10.?

File: src/components/atoms/Icon.tsx (hunk 4 above)
Comment (weeman1337) [reply src/components/atoms/Icon.tsx:31]: How can we do this? Adding something like `$alertText` or `$alertColor` that can be applied to text where the background is `$alert`. Or by having dedicated colours for the live badge: `$liveBadgeBackground` and `$liveBadgeColour`.

Comment (kerryarchibald) [reply src/components/atoms/Icon.tsx:31]: Should it not just use `primary-content` or `background`?

Comment (weeman1337) [reply src/components/atoms/Icon.tsx:31]: `primary-content` in light theme is black. That is different from the Voice Broadcast designs. `background` is the overall app background I think. Works in light mode. Doesn't match in dark mode. :shrug: 

I can put this on my list for our next sync with Gaëlle and will use a specialised colour for now.

File: src/components/views/messages/MessageEvent.tsx (hunk 5 above)
Comment (weeman1337) [reply src/components/views/messages/MessageEvent.tsx:78]: https://github.com/vector-im/element-web/issues/23354

**Review by kerryarchibald — APPROVED** [review]

**Pull request merged**